build/android/BUILD.gn

```diff
@@ -0,0 +1,340 @@
+  ]
+  deps = [
+    ":xwalk_core_library_java_app_part",
+    ":xwalk_core_library_java_library_part",
+  ]
+}
+
+action("xwalk_core_library_java_library_part") {
+  script = "//xwalk/build/android/gn/merge_xwalk_core_library_jars.py"
+  args = [
+    "--jar_type=library_part",
+    "--output_dir=$build_dir",
+  ]
+  outputs = [
+    "$root_out_dir/lib.java/xwalk_core_library_java_library_part.jar",
+  ]
+  deps = [
+    "//third_party/android_protobuf:protobuf_nano_javalib",
+    "//third_party/cardboard-java:cardboard-java",
+    "//xwalk/runtime/android/core_internal:xwalk_core_internal_java",
+    "//xwalk/runtime/android/core_internal_empty:xwalk_core_internal_empty_embedder_apk",
+  ]
+}
+
+action("xwalk_core_library_java_app_part") {
+  script = "//xwalk/build/android/gn/merge_xwalk_core_library_jars.py"
+  args = [
+    "--jar_type=app_part",
+    "--output_dir=$build_dir",
+  ]
+  outputs = [
+    "$root_out_dir/lib.java/xwalk_core_library_java_app_part.jar",
+  ]
+  deps = [
+    "//xwalk/runtime/android/core:xwalk_core_java",
+  ]
+}
+
+action("xwalk_core_library_aar") {
+  script = "//xwalk/build/android/gn/generate_xwalk_core_library_aar.py"
+  args = [
+    "-t",
+    "./",
+  ]
+  outputs = [
+    "$root_out_dir/xwalk_core_library.aar",
+  ]
+  deps = [
+    ":xwalk_core_library",
+    ":xwalk_core_library_pom_gen",
+    "//xwalk/runtime/android/core_internal_empty:xwalk_core_empty_embedder_apk",
+  ]
+}
+
+action("xwalk_shared_library_aar") {
+  script = "//xwalk/build/android/gn/generate_xwalk_core_library_aar.py"
+  args = [
+    "-t",
+    "./",
+    "--shared",
+  ]
+  outputs = [
+    "$root_out_dir/xwalk_shared_library.aar",
+  ]
+  deps = [
+    ":xwalk_shared_library",
+    ":xwalk_shared_library_pom_gen",
+    "//xwalk/runtime/android/core_internal_empty:xwalk_core_empty_embedder_apk",
+  ]
+}
+
+# Remove "build/android/maven_pom.gypi" and call "build/util/version.py" directly
+action("xwalk_core_library_pom_gen") {
+  script = "//build/util/version.py"
+  args = [
+    "-i",
+    "$src_dir/xwalk/runtime/android/maven/xwalk_core_library.pom.xml.in",
+    "-o",
+    "./xwalk_core_library.pom.xml",
+    "-e",
+    "ARTIFACT_ID='$xwalk_core_library_artifact_id'",
+    "-e",
+    "ARTIFACT_VERSION='$xwalk_version'",
+  ]
+  outputs = [
+    "$root_out_dir/xwalk_core_library.pom.xml",
+  ]
+}
+
+action("xwalk_shared_library_pom_gen") {
+  script = "//build/util/version.py"
+  args = [
+    "-i",
+    "$src_dir/xwalk/runtime/android/maven/xwalk_shared_library.pom.xml.in",
+    "-o",
+    "./xwalk_shared_library.pom.xml",
+    "-e",
+    "ARTIFACT_ID='$xwalk_shared_library_artifact_id'",
+    "-e",
+    "ARTIFACT_VERSION='$xwalk_version'",
+  ]
+  outputs = [
+    "$root_out_dir/xwalk_shared_library.pom.xml",
+  ]
+}
+
+# TODO(heke123): do we really need the core_internal_empty to get R.class files?
+# Is it possible to get R.class files from core_shell or core_internal_shell?
```

Comment:
sorry, it is a reminder to myself,  I'll delete this comments, and have a whole check on the usage of core_internal_empty_embedder and core_empty_embedder.